### PR TITLE
Update with 2019 data (2020 version)

### PR DESCRIPTION
**Pull request by SiggyF**
- updated data with 2019 (available since 2020-11-11)
- move some shared code to separate python files (can be moved to a package once collected).
- updates for latest packages (windrose, pyproj x/y -> y/x swap in pyproj > 3)

### CODE REVIEW

File: lib/psmsl.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+def missing2nan(value, missing=-99999):
+    """convert the value to nan if the float of value equals the missing value"""
+    value = float(value)
+    if value == missing:
+        return np.nan
+    return value
+
+def year2date(year_fraction, dtype='datetime64[s]'):
+    """convert a fraction of a year + fraction of a year to a date, for example 1993.12 -> 1993-02-01.
```
Comment (CindyvdVries) [inline lib/psmsl.py:16]: fraction of a year + fraction of a year -> wut?

File: lib/psmsl.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        sep=';',
+        names=('year', 'height', 'interpolated', 'flags'),
+        converters={
+            "year": lambda x: float(x),
+            "height": lambda x: missing2nan(x),
+            "interpolated": str.strip,
+        }
+    )
+    df['station'] = station.name
+    df['t'] = year2date(df.year)
```
Comment (CindyvdVries) [inline lib/psmsl.py:51]: t?

File: lib/models.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
+import numpy as np
+import statsmodels.api as sm
+
+# define the statistical model
+def linear_acceleration_model(df):
```
Comment (CindyvdVries) [inline lib/models.py:5]: bit more comments

File: lib/models.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    y = df['height']
+    X = np.c_[
+        df['year']-1970,
+        (df['year'] - 1970) * (df['year'] - 1970),
+        np.cos(2*np.pi*(df['year']-1970)/18.613),
```
Comment (CindyvdVries) [inline lib/models.py:10]: 18.613?

File: lib/models.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    fit = model.fit()
+    return fit, names
+
+
+# define the statistical model
+def quadratic_model(df, with_wind=True):
+    """This model computes a parabolic linear fit. This corresponds to the hypothesis that sea-level is accelerating."""
+    y = df['height']
+    X = np.c_[
+        df['year']-1970,
```
Comment (CindyvdVries) [inline lib/models.py:26]: there is quite some repitition in these models? maybe a seperate function to do this part per model?

**Review by CindyvdVries — APPROVED** [review]
Looks good

File: lib/psmsl.py (hunk 1 above)
Comment (SiggyF) [reply lib/psmsl.py:16]:
```suggestion
    """convert a fraction of a year + fraction of a year to a date, for example 1993.083 -~> 1993-02-01.
```

File: lib/models.py (hunk 4 above)
Comment (SiggyF) [reply lib/models.py:10]: See https://meridian.allenpress.com/jcr/article-abstract/28/2/511/27144

File: lib/psmsl.py (hunk 2 above)
Comment (SiggyF) [reply lib/psmsl.py:51]: t as in time.

File: lib/models.py (hunk 3 above)
Comment (SiggyF) [reply lib/models.py:5]: added

File: lib/models.py (hunk 5 above)
Comment (SiggyF) [reply lib/models.py:26]: I am in the process of restructuring this. For now I have to do it like this. I will update this in a next release.

**Pull request merged**